### ISSUE TITLE
Categories for Events

### ISSUE THREAD

**Issue by svenseeberg** [Integreat/cms-django#273]
Categories could be useful to enable filtering of events in the App

**Comment by JoeyStk** [comment]
Do we need also specific colors and icons, like we had with POIs?

**Comment by timobrembeck** [comment]
> @MizukiTemma and I were wondering if we needed also specific colors and icons, like we had with POIs?

I think without the restriction of having the map pins, we don't necessarily have to hardcode these options, but could also provide a hex color input and the option to use icons from the media library.
On the other hand, if we want to copy as much code as possible from the POI categories, we could as well use the same mechanism of hardcoding colors and icons.

**Comment by JoeyStk** [comment]
So far this is only an idea and there are a few things that need to be addressed before we can work on this:

- We need the UI/UX workflow on how to go about this (like is this similar to the POI-categories or should be different)?
- Who should be able to create these categories?

**Comment by steffenkleinle** [comment]
The proposed categories have been posted here by the @digitalfabrik/service-team: https://github.com/digitalfabrik/integreat-app/issues/2707#issuecomment-2717555704
> - ~~Location (Ort) - especially important in Landkreisen to filter for the single locations (Orte)~~
- Kids & Family
- Sport
- Health
- Women
- Senior citizens
- Language
- Leisure
- Politics
- Arts & Culture
- Integration
- Markets
- Religion
- Career and further education
- Consulting services
- Others

**Comment by MizukiTemma** [comment]
> * We need the UI/UX workflow on how to go about this (like is this similar to the POI-categories or should be different)?
> * Who should be able to create these categories?

I suggest to implement this in the same way with POI-category: same workflow (form, view, etc), same permission setting.

A question is, whether then the field "category" shuold be mandatory for every event.

**Comment by steffenkleinle** [comment]
> > * We need the UI/UX workflow on how to go about this (like is this similar to the POI-categories or should be different)?
> > * Who should be able to create these categories?
> 
> I suggest to implement this in the same way with POI-category: same workflow (form, view, etc), same permission setting.

👍 

> A question is, whether then the field "category" shuold be mandatory for every event.

I'd say yes please 👍

**Comment by hauf-toni** [comment]
fyi: these categories are not final yet, just a proposal. I will review them and send you my feedback asap.

**Comment by MizukiTemma** [comment]
@hauf-toni @steffenkleinle 
From CMS perspective: Icons are hard-coded but categories can be always created and edited.

**Comment by steffenkleinle** [comment]
> [@hauf-toni](https://github.com/hauf-toni) [@steffenkleinle](https://github.com/steffenkleinle) From CMS perspective: Icons are hard-coded but categories can be always created and edited.

Sounds good to me. Would be cool if the workflow would be the same as for poi categories and we get them normally delivered (incl. translations) in the events endpoint.

Does it perhaps make sense to share categories between pois and events, i.e. use the same categories?

**Comment by hauf-toni** [comment]
To enhance the user experience + improve event discoverability, I have refined the event categories based on best practices in information architecture and usability design. The previous list contained overlapping or very specific categories, which could make navigation cumbersome (such a great english word, right? thanks deepl!) for users.

💬 **Proposed Final List (9-10 Categories)**

Kids & Family
Leisure & Sport (combining leisure and sports)
Health
Community & Social Affairs (merging Women, Seniors, Politics, and Integration)
Language
Arts & Culture
Markets
Religion
Career & Education (merging Career and Further Education + Consulting Services)

The categorization should be reviewed periodically based on user feedback and evolving needs.

What do you think @laravoelk ?

**Comment by laravoelk** [comment]
I think that this looks good, thank you. It is just very important for us that the category locations (Orte) will not be forgotten, as this is the most requested category of our municipalities. But I believe this was put into a separate issue, right?

**Comment by MizukiTemma** [comment]
@hauf-toni 
As soon as icons for the initial categories are provided, this issue is ready to be tackled.

**Comment by dkehne** [comment]
How do we deal with events/dates from external sources that deliver categories to us? It's plural because one external event can have multiple categories. 

Idea 1: On the external calendar management page (where an external calendar is added) one has to select an available category which are then all external events from the targeted source are assigned. that can be repeated multiple times because through our filter function there, one can put the external calendar multiple times in there and map each entry with a filtered external category on our own categories.

Idea 2: We show up the categories as named as in the external calendar. The we need to solve the translation issue of those categories.

Idea 3: We put all events into a category called "external".

I am a huge fan of idea 1, mapping one source entry to one of our categories.
